xarray/core/dataarray.py

```diff
@@ -1095,6 +1095,31 @@ def to_masked_array(self, copy=True):
         isnull = pd.isnull(self.values)
         return np.ma.MaskedArray(data=self.values, mask=isnull, copy=copy)
 
+    def to_netcdf(self, *args, **kwargs):
+        """
+        Write DataArray contents to a netCDF file.
+
+        Parameters
+        ----------
+        All parameters are passed directly to `xarray.Dataset.to_netcdf`.
+
+        Notes
+        -----
+        Only xarray.Dataset objects can be written to netCDF files, so
+        the xarray.DataArray is converted to a xarray.Dataset object
+        containing a single variable. If the DataArray has no name, then
+        it is given the name 'data'.
+        """
+        if not self.name and self.name not in list(self.coords):
+            # If the name isn't blank/None or the same as one of the coords
+            # (as the latter is invalid for a netCDF file) then give it a name
+            # of `data`
+            dataset = self.to_dataset(name='data')
```

Comment:
To be extra thorough, we could save the original `name` as an attribute (e.g., `"__xarray_dataarray_name__"`) if it matches a coordinate, and then use that to set the name on the result. That could ensure that we always faithfully roundtrip the DataArray exactly.